### PR TITLE
Modulo 4: Implement Clean Architecture

### PR DESCRIPTION
**Pull request by jere53**
The project structure has been reorganized to comply with Clean Architecture, following [this](https://github.com/MarcosCGlobant/flutter_clean_architecture_example) folder structure. Flutter Lint package was also added, and strict rule package followed

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -9,8 +10,13 @@ class MyApp extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final MovieRepository repository = MovieRepository();
```
Comment (matiasilveiro-gl) [inline lib/main.dart:13]: This should be `final IMovieRepository repository = MovieRepository();`!
The presentation layer uses abstractions from the domain layer

File: lib/src/presentation/view/movies_view.dart [hunk 2]
```diff
@@ -1,15 +1,16 @@
 import 'package:flutter/material.dart';
-import 'package:movie_app_jsalinas/models/genre_model.dart';
-import '../../../pages/details_page.dart';
-import '../../../utils/i_movie_repository.dart';
-import '../../../widgets/movie_poster.dart';
-import '../../../widgets/movie_score.dart';
-import '../../../utils/ui_constants.dart';
-import '../../../models/movie.dart';
-import '../models/genre.dart';
+import 'package:movie_app_jsalinas/src/domain/repository/i_movie_repository.dart';
```
Comment (matiasilveiro-gl) [inline lib/src/presentation/view/movies_view.dart:2]: Relative import, to keep consistency

**Review by matiasilveiro-gl — APPROVED** [review]
LGTM! Minor comments

File: lib/main.dart (hunk 1 above)
Comment (jere53) [reply lib/main.dart:13]: fixed! Also moved the instantiation of the MovieRepository to the main function, so MyApp widget no longer depends on that implementation. This means I can pass a different repo implementation to MyApp in yest/widget.test.dart (I pass the same implementation for now, though)

**Review by matiasilveiro-gl — APPROVED** [review]

File: lib/main.dart [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
+  runApp(
+    MyApp(
+      repository: repository,
+    ),
+  );
+}
 
 class MyApp extends StatelessWidget {
-  const MyApp({super.key});
+  const MyApp({super.key, required this.repository});
```
Comment (Kevin-Guillen) [inline lib/main.dart:18]: commaa

File: test/widget_test.dart [hunk 4]
```diff
@@ -9,11 +9,13 @@ import 'package:flutter/material.dart';
...
 
 import 'package:movie_app_jsalinas/main.dart';
+import 'package:movie_app_jsalinas/src/data/repository/movie_repository.dart';
 
 void main() {
   testWidgets('Counter increments smoke test', (WidgetTester tester) async {
     // Build our app and trigger a frame.
-    await tester.pumpWidget(const MyApp());
+    final repository = MovieRepository();
+    await tester.pumpWidget(MyApp(repository: repository,));
```
Comment (Kevin-Guillen) [inline test/widget_test.dart:18]: format code

Comment (jere53) [reply test/widget_test.dart:18]: fixed

File: lib/src/data/repository/movie_repository.dart [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    final Map<String, dynamic> decodedGenreData =
+        jsonDecode(genreData) as Map<String, dynamic>;
+
+    final List<Map<String, dynamic>> genreList =
+        List.from(decodedGenreData['genres'] as Iterable);
+
+    final List<Genre> genres = GenreModel.fromJson(
+      genreList,
+    )!
+        .toList();
```
Comment (FranciscoGaraventa) [inline lib/src/data/repository/movie_repository.dart:35]: Why you need to force this nullability on the fromJson

File: lib/src/domain/entity/genre.dart [hunk 6]
```diff
@@ -9,8 +9,8 @@ class Genre {
 
   factory Genre.fromJson(Map<String, dynamic> json) {
     return Genre(
-      json['id'],
-      json['name'],
+      json['id'] as int,
+      json['name'] as String,
```
Comment (FranciscoGaraventa) [inline lib/src/domain/entity/genre.dart:13]: This cast is necessary?

Comment (jere53) [reply lib/src/domain/entity/genre.dart:13]: the Linter strict rules enforce it

File: lib/src/data/repository/movie_repository.dart (hunk 5 above)
Comment (jere53) [reply lib/src/data/repository/movie_repository.dart:35]: hmm, i dont actually. Changed fromJson to use non-nullable types

File: lib/src/domain/entity/genre.dart (hunk 6 above)
Comment (jere53) [reply lib/src/domain/entity/genre.dart]: fixed

**Review by FranciscoGaraventa — APPROVED** [review]
LG

**Pull request merged**